src/JuMP.jl

```diff
@@ -210,38 +211,44 @@ function mode(model::Model)
     end
 end
 
-# Needs to defined by all `AbstractModel`
-num_variables(m::Model) = MOI.get(m, MOI.NumberOfVariables())
+"""
+    num_variables(model::Model)
 
+Returns number of variables in `model`.
 """
-    numnlconstr(m::Model)
+num_variables(model::Model) = MOI.get(model, MOI.NumberOfVariables())
 
-returns the number of nonlinear constraints associated with the `Model m`
 """
-numnlconstr(m::Model) = m.nlpdata !== nothing ? length(m.nlpdata.nlconstr) : 0
+    numnlconstr(model::Model)
 
+Returns the number of nonlinear constraints associated with the `model`.
+"""
+function numnlconstr(model::Model)
+    return model.nlpdata !== nothing ? length(model.nlpdata.nlconstr) : 0
+end
 
 """
-    objectivebound(m::Model)
+    objectivebound(model::Model)
 
-Return the best known bound on the optimal objective value after a call to `solve`.
+Return the best known bound on the optimal objective value after a call to
+`optimize(model)`.
 """
-objectivebound(m::Model) = MOI.get(m, MOI.ObjectiveBound())
+objectivebound(model::Model) = MOI.get(model, MOI.ObjectiveBound())
 
 """
-    objectivevalue(m::Model)
+    objectivevalue(model::Model)
 
-Return the objective value after a call to `solve`.
+Return the objective value after a call to `optimize(model)`.
 """
-objectivevalue(m::Model) = MOI.get(m, MOI.ObjectiveValue())
+objectivevalue(model::Model) = MOI.get(model, MOI.ObjectiveValue())
```

Comment:
We haven't yet moved over the naming discussion to the style guide, but what do you think about `objective_value` etc for these?

Reply:
I'm a fan of underscores (another difference from Base, but w/e...), so sgtm

Reply:
Let's hold off on renaming until we resolve the style question, otherwise it might just be unnecessary churn.